guide/src/building_and_distribution.md

```diff
@@ -1,58 +1,91 @@
 # Building and Distribution
 
-## Python version
+This chapter of the guide goes into detail on how to build and distribute projects using PyO3. The way to achieve this is very different depending on whether the project is a Python module implemented in Rust, or a Rust binary embedding Python. For both types of project there are also common problems such as the Python version to build for and the [linker](https://en.wikipedia.org/wiki/Linker_(computing)) arguments to use.
+
+The material in this chapter is intended for users who have already read the PyO3 [README](#index.md). It covers in turn the choices that can be made for Python modules and for Rust binaries. There is also a section on the end about cross-compiling projects using PyO3.
+
+There is an additional sub-chapter dedicated to [supporting multiple Python versions](./building_and_distribution/multiple_python_versions.html).
+
+## Configuring the Python version
+
+PyO3 uses a build script (backed by the [`pyo3-build-config`] crate) to determine the Python version and set the correct linker arguments. By default it will attempt to use the following in order:
+ - Any active Python virtualenv.
+ - The `python` executable (if it's a Python 3 interpreter).
+ - The `python3` executable.
+
+You can override the Python interpreter by setting the `PYO3_PYTHON` environment variable, e.g. `PYO3_PYTHON=python3.6`, `PYO3_PYTHON=/usr/bin/python3.9`, or even a PyPy interpreter `PYO3_PYTHON=pypy3`.
+
+Once the Python interpreter is located, `pyo3-build-config` executes it to query the information in the `sysconfig` module which is needed to configure the rest of the compilation.
+
+To validate the configuration which PyO3 will use, you can run the `print-config` binary from the `pyo3-build-config` crate. An example usage of this binary is shown below:
```

Comment:
Could `print-config` be useful as a feature of `pyo3`, so that you can get this output simply by activating it within Cargo.toml, avoiding the need to clone pyo3?

Reply:
Hmm interesting idea. I guess there could be a `PYO3_PRINT_CONFIG=1` environment variable which prints the config and then stops the build? I'm not aware of a way to make the output be printed and successfully compile...

Reply:
Right, an env var is probably even easier. And yes, I also think it needs to error out in order to see stdout/stderr.

Reply:
Added a commit to remove the `print-config` binary and put it behind an environment variable instead. Feels really nice!